src/test/java/seedu/exercise/logic/commands/events/ResolveEventTest.java

```diff
@@ -0,0 +1,75 @@
+package seedu.exercise.logic.commands.events;
+
+import static seedu.exercise.logic.commands.events.EventTestUtil.LEVEL_ONE_AND_TWO_REGIME_DATE_1_CONFLICT;
+import static seedu.exercise.logic.commands.events.EventTestUtil.LEVEL_ONE_REGIME_DATE_1_SCHEDULE;
+import static seedu.exercise.logic.commands.events.EventTestUtil.LEVEL_THREE_REGIME;
+import static seedu.exercise.logic.commands.events.EventTestUtil.LEVEL_THREE_REGIME_DATE_1_SCHEDULE;
+import static seedu.exercise.logic.commands.events.EventTestUtil.LEVEL_TWO_REGIME_DATE_1_SCHEDULE;
+import static seedu.exercise.logic.commands.events.EventTestUtil.RESOLVE_LEVEL_ONE_AND_LEVEL_TWO_CONFLICT_AND_CREATE_LEVEL_THREE_PAYLOAD;
+import static seedu.exercise.logic.commands.events.EventTestUtil.RESOLVE_LEVEL_ONE_AND_LEVEL_TWO_CONFLICT_AND_TAKE_LEVEL_TWO_PAYLOAD;
+import static seedu.exercise.logic.commands.events.EventTestUtil.assertRedoEventSuccess;
+import static seedu.exercise.logic.commands.events.EventTestUtil.assertUndoEventSuccess;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import seedu.exercise.MainApp;
+import seedu.exercise.commons.core.State;
+import seedu.exercise.model.Model;
+import seedu.exercise.model.ModelManager;
+
+public class ResolveEventTest {
+    private Model actualModel;
+    private Model expectedModel;
+    private ResolveEvent resolveEvent;
+
+    @BeforeEach
+    public void setUp() {
+        MainApp.setState(State.IN_CONFLICT);
```

Comment:
does the app's state have to be `IN_CONFLICT` in order for the tests to work? Because, `undo` command should be executed when the state is `NORMAL`

Reply:
You are right, it is not needed for executing undo or redo. Removed it already.